### PR TITLE
bhyve branded zone, v2

### PR DESCRIPTION
**Pull request by citrus-it**
This updates the bhyve branded zone.

The existing brand loopback mounts the zone's XML file into the zone's `/etc`, then a python init script parses it and starts bhyve with the appropriate arguments, and restarts it again if it exits due to a VM reboot.

A v1 zone has a process tree like this:

```
17542  zsched
  17622  /usr/bin/python3 /usr/lib/brand/bhyve/init
    1682   /usr/sbin/bhyve -U 15d75ebd-b708-c0ef-eaa1-8670b7ce8a40 -H -B 1,manu...
```

The v2 brand takes advantage of the recently added bhyve configuration file. During zone boot, a hook runs in the GZ that parses the zone XML file, and generates a bhyve configuration file and places it in the zone root.
The zone's init is a small C program which sets a few things up and then execs the real bhyve, setting argv[0] to include the zone name.
When bhyve exits due to a VM reboot, the zone brand restarts it, without destroying and re-creating the VM, which makes for much faster reboots.

A v2 zone has a process tree that looks like this:

```
19236  zsched
  19318  bhyve-bhyvetest -k /etc/bhyve.cfg
```

### CODE REVIEW

File: src/brand/bhyve/boot [hunk 1]
```diff
@@ -309,45 +369,59 @@ for dev in root.findall('./attr[@name]'):
...
+
+writecfg(fh, '#\n# Generated from zone configuration\n#')
+
+for line in p.stdout.splitlines():
+    if line.startswith('config.dump'): continue
+    writecfg(fh, line)
+
+if not testmode:
+    tf = fh.name
+    fh.close()
```
Comment (sjorge) [inline src/brand/bhyve/boot:418]: Should we not excplicitly close the fine in test mode too? I know Python closes it at end of execution, but being explicit seems cleaner and safer

File: src/man/bhyve.5 [hunk 2]
```diff
@@ -180,10 +180,15 @@ presented to the guest.
...
 .Bl -bullet
 .It
-.Sy virtio-net-viona Pq default
+.Sy virtio-net-viona Pq accelerated virtio interface, default
+.It
+.Sy virtio-net Pq legacy virtio interface
 .It
 .Sy e1000
 .El
+.Pp
```
Comment (sjorge) [inline src/man/bhyve.5:189]: Should we also mention that only the legacy virtio-net will support promisc? That should help people that want to try and run things like VRRP/CARP/… in the guest.

File: src/brand/bhyve/boot (hunk 1 above)
Comment (citrus-it) [reply src/brand/bhyve/boot:418]: If we're in test mode, the file was never opened (fh is `None`)

File: src/man/bhyve.5 (hunk 2 above)
Comment (citrus-it) [reply src/man/bhyve.5:189]: That isn't there yet. We need 13763 and 13738 first.

Comment (sjorge) [reply src/man/bhyve.5:189]: So e1000 works without those too?

Comment (citrus-it) [reply src/man/bhyve.5:189]: No, you can attach them to the VM, and they will appear as PCI devices, but they will not work.

Comment (sjorge) [reply src/man/bhyve.5:189]: Ack

**Review by hadfl — DISMISSED** [review]

**Review by hadfl — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Pull request merged**